src/components/data-table/data_table.ts

```diff
@@ -79,17 +81,38 @@ export class MdDataTable implements AfterContentInit {
       .map((tr: MdDataTableSelectableRow) => tr.selectableValue);
   }
 
+  _unsubscribeChildren() {
+    this.selected = [];
+    if (this._listeners.length) {
```

Comment:
you shouldn't need to check .length here.  If there are 0 items in the array, the forEach will be a no-op.